components/director/pkg/graphql/formation_template.go

```diff
@@ -11,6 +11,7 @@ type FormationTemplate struct {
 	Webhooks               []*Webhook    `json:"webhooks"`
 	LeadingProductIDs      []string      `json:"leadingProductIDs"`
 	SupportsReset          bool          `json:"supportsReset"`
+	DiscoveryConsumers     []string      `json:"discoveryConsumers"`
```

Comment:
In the corresponding validation file: `formation_template_validation.go` we can check if the values of the discovery consumers are among the application/runtime types:

Example check:
`validation.Field(&i.DiscoveryConsumers, validation.In(i.RuntimeTypes, i.ApplicationTypes)),`

If you find it necessary, you can extend the unit tests in the `formation_template_validation_test.go`